tests/spec/rpc_01-system_test.go

```diff
@@ -0,0 +1,140 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package spec
+
+import (
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/rpc/modules"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/require"
+
+	rpc "github.com/ChainSafe/gossamer/tests/rpc"
+)
+
+func TestSystemRPC(t *testing.T) {
+	testsCases := []struct {
+		description string
+		method      string
+		expected    interface{}
+		skip        bool
+	}{
+		{ //TODO
+			description: "test system_name",
+			method:      "system_name",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_version",
+			method:      "system_version",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_chain",
+			method:      "system_chain",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_properties",
+			method:      "system_properties",
+			skip:        true,
+		},
+		{
+			description: "test system_health",
+			method:      "system_health",
+			expected: modules.SystemHealthResponse{
+				Health: common.Health{
+					Peers:           2,
+					IsSyncing:       false,
+					ShouldHavePeers: true,
+				},
+			},
+			skip: false,
+		},
+		{
+			description: "test system_peers",
+			method:      "system_peers",
+			expected: modules.SystemPeersResponse{
+				Peers: []common.PeerInfo{},
+			},
+			skip: false,
+		},
+		{
+			description: "test system_network_state",
+			method:      "system_networkState",
+			expected: modules.SystemNetworkStateResponse{
+				NetworkState: common.NetworkState{
+					PeerID: "",
+				},
+			},
+			skip: false,
+		},
+		{ //TODO
+			description: "test system_addReservedPeer",
+			method:      "system_addReservedPeer",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_removeReservedPeer",
+			method:      "system_removeReservedPeer",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_nodeRoles",
+			method:      "system_nodeRoles",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_accountNextIndex",
+			method:      "system_accountNextIndex",
+			skip:        true,
+		},
+	}
+
+	t.Log("going to Bootstrap Gossamer node")
+
+	localPidList, err := rpc.StartNodes(t, make([]*exec.Cmd, 1))
+
+	//use only first server for tests
+	require.Nil(t, err)
+
+	time.Sleep(time.Second) // give server a second to start
+
+	for _, test := range testsCases {
+		t.Run(test.description, func(t *testing.T) {
+			if test.skip {
+				t.Skip("RPC endpoint not yet implemented")
+				return
+			}
+
+			respBody, err := rpc.PostRPC(t, test.method, "http://"+rpc.GOSSAMER_NODE_HOST+":"+currentPort, "{}")
+			require.Nil(t, err)
+
+			target := rpc.DecodeRPC(t, respBody, test.method)
+
+			require.NotNil(t, target)
```

Comment:
where is the check for if result == expected?

Reply:
Yes, I added the logic to it now... its for now duplicating a bit with `system_integration_test.go`. This is ok since system_integration_test will be updated soon and follow other directions for testing with stable pre-saved gold files